Apps/Examples/Examples/Models/ExampleProtocol.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+import MapboxMaps
+
+public protocol ExampleProtocol: AnyObject {
+    func resourceOptions() -> ResourceOptions
+    func finish()
+}
+
+extension ExampleProtocol {
+    public func resourceOptions() -> ResourceOptions {
+        guard let accessToken = AccountManager.shared.accessToken else {
+            fatalError("Access token not set")
+        }
+
+        guard !accessToken.isEmpty else {
+            fatalError("Empty access token")
+        }
+
+        let resourceOptions = ResourceOptions(accessToken: accessToken)
+        return resourceOptions
+    }
+
+    public func finish() {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 3.0) {
+            NotificationCenter.default.post(name: Example.finishNotificationName, object: self)
+        }
+    }
+}
```

Comment:
The previous code was added to ensure that if finish is called multiple times, that the notification is still only posted once. Is that still a concern with the refactor?

Reply:
It shouldn't matter anymore because the test is now looking for the notification from a specific object.

Reply:
…so finish notifications from one test shouldn't impact any other tests (unless somehow the view controllers get allocated at the same memory address, in which case the [object comparison](https://developer.apple.com/documentation/foundation/nsnotification) would not help)

> The objects of a notification are compared using pointer equality for local notifications.